postman_problems/shortest_path_mod.py

```diff
@@ -0,0 +1,2184 @@
+# -*- coding: utf-8 -*-
+#    Copyright (C) 2004-2019 by
+#    Aric Hagberg <hagberg@lanl.gov>
+#    Dan Schult <dschult@colgate.edu>
+#    Pieter Swart <swart@lanl.gov>
+#    All rights reserved.
+#    BSD license.
+#
+# Authors:  Aric Hagberg <hagberg@lanl.gov>
+#           Loïc Séguin-C. <loicseguin@gmail.com>
+#           Dan Schult <dschult@colgate.edu>
+#           Niels van Adrichem <n.l.m.vanadrichem@tudelft.nl>
+# Modified by:
+#           Jack Connor <jconnor@baaqmd.gov>
```

Comment:
This is a ton of code.  Is any of it in a library that can be imported?  Looks like some/most of it might be networkx direcrtly?

Reply:
This is a direct copy of networkx code with a modification of mine deep within. I had to keep the whole file because the code is written as nested functions, and the part I had to modify was pretty deep in there. The modification allowed for turns to be weighted in the dijkstra paths algorithm.